app/src/main/java/com/pileproject/drive/execution/CarControllerBase.java

```diff
@@ -0,0 +1,302 @@
+     */
+    public int getSoundSensorDb() {
+        if (mSoundSensor == null) return -1;
+        return mSoundSensor.getDb();
+    }
+
+    /**
+     * Checks the touch sensor is touched or not now.
+     *
+     * @return is touched (true) or not (false)
+     */
+    public boolean isTouchSensorTouched() {
+        if (mTouchSensor == null) return false;
+        return mTouchSensor.isTouched();
+    }
+
+    public int getTouchSensorTouchedCount() {
+        if (mTouchSensor == null) return -1;
+        return mTouchSensor.getTouchedCount();
+    }
+
+
+    /**
+     * move forward
+     * public field variables leftMotorPower and rightMotorPower are set as
+     * motor's power
+     */
+    public void moveForward() {
+        move(MotorDir.Forward, MotorDir.Forward);
+    }
+
+    /**
+     * move backward
+     * public field variables leftMotorPower and rightMotorPower are set as
+     * motor's power
+     */
+    public void moveBackward() {
+        move(MotorDir.Backward, MotorDir.Backward);
+    }
+
+    /**
+     * turn left
+     * motor's powers are set to the constant power
+     */
+    public void turnLeft() {
+        move(MotorDir.Backward, MotorDir.Forward);
+    }
+
+    /**
+     * turn right
+     * motor's powers are set to the constant power
+     */
+    public void turnRight() {
+        move(MotorDir.Forward, MotorDir.Backward);
+    }
+
+    /**
+     * Drives the both motors toward specified direction
+     *
+     * @param leftMotorDir
+     * @param rightMotorDir
+     */
+    private void move(MotorDir leftMotorDir, MotorDir rightMotorDir) {
+        if (mLeftMotor != null) {
+            switch (leftMotorDir) {
+                case Forward:
+                    mLeftMotor.setSpeed(mLeftMotorPower);
+                    mLeftMotor.forward();
+                    break;
+                case Neutral:
+                    mLeftMotor.stop();
+                    break;
+                case Backward:
+                    mLeftMotor.setSpeed(mLeftMotorPower);
+                    mLeftMotor.backward();
+                    break;
+            }
+        }
+
+        if (mRightMotor != null) {
+            switch (rightMotorDir) {
+                case Forward:
+                    mRightMotor.setSpeed(mRightMotorPower);
+                    mRightMotor.forward();
+                    break;
+                case Neutral:
+                    mRightMotor.stop();
+                    break;
+                case Backward:
+                    mRightMotor.setSpeed(mRightMotorPower);
+                    mRightMotor.backward();
+                    break;
+            }
+        }
+    }
+
+    /**
+     * A setter of motor power.
+     *
+     * @param kind see {@link MotorKind}
+     * @param power [0, 100]
+     *              If an out of bound value is passed, then the value will be clipped
+     *              (e.g., -10 to 0, 200 to 100).
+     */
+    public void setMotorPower(MotorKind kind, int power) {
```

Comment:
I prefer `percent` to `power` because it's clearer to understand the range of the value.